pkg/util/core.go

```diff
@@ -0,0 +1,32 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+// +build !windows
+
+package util
+
+import (
+	"fmt"
+	"runtime/debug"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"golang.org/x/sys/unix"
+)
+
+// SetupCoreDump enables core dumps and sets the core dump size limit based on configuration
+func SetupCoreDump() error {
+	if config.Datadog.GetBool("go_core_dump") {
+		debug.SetTraceback("crash")
+
+		err := unix.Setrlimit(unix.RLIMIT_CORE, &unix.Rlimit{
```

Comment:
Do the agents have always the right to perform this operation?

Reply:
They may not have the right to do it if the system has hard user limits on core dump sizes. In that case the `Setrlimit` function will return an error that will be logged, so the user/dev can modify that hard limit. There is not much else we could do in that situation.